### ISSUE TITLE
The ODGI graph does not pass the handle graph tests ported over from VG

### ISSUE THREAD

**Issue by jeizenga** [vgteam/libbdsg#19]
I haven't looked into this deeply yet, but we should figure out whether this is the fault of the tests or the implementation, or perhaps a disagreement over the generality of our formalism.

**Comment by ekg** [comment]
Odgi test calls a version of these tests from a few months ago. What failed?

**Comment by jeizenga** [comment]
Okay, I think I at least have a clue what's happening. It relates to this line in the orginal `node.cpp`:
https://github.com/vgteam/odgi/blob/master/src/node.cpp#L36

I think I wrote to you about this on slack a while back. The issue is that the assert statement was never actually executing because of this line in the CMakeLists, which removes all asserts
https://github.com/vgteam/odgi/blob/master/CMakeLists.txt#L235

I had to edit the line to have `edge_count()` instead of `edge_count` in order to compile. However, this assert statement has probably never actually run in the tests in the odgi repository because of that cmake flag. According to the slack messages I left you, you should be able to reproduce this from the `vg view` GFA from `tiny.vg`.

**Comment by ekg** [comment]
I get it now. I forgot that release removed asserts.

This assertion is incorrect and should be removed or changed. The length of
res should be edge_count*EDGE_RECORD_LENGTH.

**Comment by jeizenga** [comment]
Another update: if I comment out that assert everything passes. Is that assert statement correct?

**Comment by ekg** [comment]
No, as I described, the assert is not correct. It was left over from a
previous version. It just needs to be removed.

**Comment by ekg** [comment]
Ok I'll take a look. How do I get to the same state as you?
